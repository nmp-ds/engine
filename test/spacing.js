import { expect, test } from 'vitest'
import { setup } from './_helpers.js'

setup()

test('padding works', async ({ uno }) => {
  const classes = ['p-8', 'px-2', 'py-4', 'pl-32', 'pr-16', 'pb-8', 'pt-16']
  const { css } = await uno.generate(classes)
  expect(css).toMatchInlineSnapshot(`
    "/* layer: default */
    .p-8{padding:0.8rem;}
    .px-2{padding-left:0.2rem;padding-right:0.2rem;}
    .py-4{padding-top:0.4rem;padding-bottom:0.4rem;}
    .pb-8{padding-bottom:0.8rem;}
    .pl-32{padding-left:3.2rem;}
    .pr-16{padding-right:1.6rem;}
    .pt-16{padding-top:1.6rem;}"
  `)
})

test('padding skips bad values', async ({ uno }) => {
  const classes = ['p-7', 'px-3', 'py-5', 'pl-33', 'pr-19', 'pb-9', 'pt-17']
  const { css } = await uno.generate(classes)
  expect(css).toMatchInlineSnapshot('""')
})

test('margin works', async ({ uno }) => {
  const classes = ['m-8', 'mx-2', 'my-4', 'ml-32', 'mr-16', 'mb-8', 'mt-16', '-m-8', 'm-auto', 'ml-auto']
  const { css } = await uno.generate(classes)
  expect(css).toMatchInlineSnapshot(`
    "/* layer: default */
    .-m-8{margin:-0.8rem;}
    .m-8{margin:0.8rem;}
    .m-auto{margin:auto;}
    .mx-2{margin-left:0.2rem;margin-right:0.2rem;}
    .my-4{margin-top:0.4rem;margin-bottom:0.4rem;}
    .mb-8{margin-bottom:0.8rem;}
    .ml-32{margin-left:3.2rem;}
    .ml-auto{margin-left:auto;}
    .mr-16{margin-right:1.6rem;}
    .mt-16{margin-top:1.6rem;}"
  `)
})

test('margin skips bad values', async ({ uno }) => {
  const classes = ['m-7', 'mx-3', 'my-5', 'ml-33', 'mr-19', 'mb-9', 'mt-17', '-m-11']
  const { css } = await uno.generate(classes)
  expect(css).toMatchInlineSnapshot('""')
})
