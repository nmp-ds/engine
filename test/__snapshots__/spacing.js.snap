// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`arbitrary margin with css variables works 1`] = `
"/* layer: default */
.m-\\[var\\(--w-spacing\\)\\]{margin:var(--w-spacing);}
.mx-\\[var\\(--w-spacing\\)\\]{margin-left:var(--w-spacing);margin-right:var(--w-spacing);}
.my-\\[var\\(--w-spacing\\)\\]{margin-top:var(--w-spacing);margin-bottom:var(--w-spacing);}
.mb-\\[var\\(--w-spacing\\)\\]{margin-bottom:var(--w-spacing);}
.ml-\\[var\\(--w-spacing\\)\\]{margin-left:var(--w-spacing);}
.mr-\\[var\\(--w-spacing\\)\\]{margin-right:var(--w-spacing);}
.mt-\\[var\\(--w-spacing\\)\\]{margin-top:var(--w-spacing);}"
`;

exports[`arbitrary margin with warp css var and default value works 1`] = `
"/* layer: default */
.m-\\[var\\(--w-spacing\\,_8px\\)\\]{margin:var(--w-spacing, 8px);}
.mx-\\[var\\(--w-spacing\\,_8px\\)\\]{margin-left:var(--w-spacing, 8px);margin-right:var(--w-spacing, 8px);}
.my-\\[var\\(--w-spacing\\,_8px\\)\\]{margin-top:var(--w-spacing, 8px);margin-bottom:var(--w-spacing, 8px);}
.mb-\\[var\\(--w-spacing\\,_8px\\)\\]{margin-bottom:var(--w-spacing, 8px);}
.ml-\\[var\\(--w-spacing\\,_8px\\)\\]{margin-left:var(--w-spacing, 8px);}
.mr-\\[var\\(--w-spacing\\,_8px\\)\\]{margin-right:var(--w-spacing, 8px);}
.mt-\\[var\\(--w-spacing\\,_8px\\)\\]{margin-top:var(--w-spacing, 8px);}"
`;

exports[`arbitrary margin with warp tokens works 1`] = `
"/* layer: default */
.m-\\[--w-spacing\\]{margin:var(--w-spacing);}
.mx-\\[--w-spacing\\]{margin-left:var(--w-spacing);margin-right:var(--w-spacing);}
.my-\\[--w-spacing\\]{margin-top:var(--w-spacing);margin-bottom:var(--w-spacing);}
.mb-\\[--w-spacing\\]{margin-bottom:var(--w-spacing);}
.ml-\\[--w-spacing\\]{margin-left:var(--w-spacing);}
.mr-\\[--w-spacing\\]{margin-right:var(--w-spacing);}
.mt-\\[--w-spacing\\]{margin-top:var(--w-spacing);}"
`;

exports[`arbitrary margin works 1`] = `
"/* layer: default */
.m-\\[8\\]{margin:0.8rem;}
.m-\\[8px\\]{margin:8px;}
.m-\\[8rem\\]{margin:8rem;}
.mx-\\[8\\]{margin-left:0.8rem;margin-right:0.8rem;}
.mx-\\[8px\\]{margin-left:8px;margin-right:8px;}
.mx-\\[8rem\\]{margin-left:8rem;margin-right:8rem;}
.my-\\[8\\]{margin-top:0.8rem;margin-bottom:0.8rem;}
.my-\\[8px\\]{margin-top:8px;margin-bottom:8px;}
.my-\\[8rem\\]{margin-top:8rem;margin-bottom:8rem;}
.mb-\\[8\\]{margin-bottom:0.8rem;}
.mb-\\[8px\\]{margin-bottom:8px;}
.mb-\\[8rem\\]{margin-bottom:8rem;}
.ml-\\[8\\]{margin-left:0.8rem;}
.ml-\\[8px\\]{margin-left:8px;}
.ml-\\[8rem\\]{margin-left:8rem;}
.mr-\\[8\\]{margin-right:0.8rem;}
.mr-\\[8px\\]{margin-right:8px;}
.mr-\\[8rem\\]{margin-right:8rem;}
.mt-\\[8\\]{margin-top:0.8rem;}
.mt-\\[8px\\]{margin-top:8px;}
.mt-\\[8rem\\]{margin-top:8rem;}"
`;

exports[`arbitrary padding with css variables works 1`] = `
"/* layer: default */
.p-\\[var\\(--w-spacing\\)\\]{padding:var(--w-spacing);}
.px-\\[var\\(--w-spacing\\)\\]{padding-left:var(--w-spacing);padding-right:var(--w-spacing);}
.py-\\[var\\(--w-spacing\\)\\]{padding-top:var(--w-spacing);padding-bottom:var(--w-spacing);}
.pb-\\[var\\(--w-spacing\\)\\]{padding-bottom:var(--w-spacing);}
.pl-\\[var\\(--w-spacing\\)\\]{padding-left:var(--w-spacing);}
.pr-\\[var\\(--w-spacing\\)\\]{padding-right:var(--w-spacing);}
.pt-\\[var\\(--w-spacing\\)\\]{padding-top:var(--w-spacing);}"
`;

exports[`arbitrary padding with warp css var and default value works 1`] = `
"/* layer: default */
.p-\\[var\\(--w-spacing\\,_8px\\)\\]{padding:var(--w-spacing, 8px);}
.px-\\[var\\(--w-spacing\\,_8px\\)\\]{padding-left:var(--w-spacing, 8px);padding-right:var(--w-spacing, 8px);}
.py-\\[var\\(--w-spacing\\,_8px\\)\\]{padding-top:var(--w-spacing, 8px);padding-bottom:var(--w-spacing, 8px);}
.pb-\\[var\\(--w-spacing\\,_8px\\)\\]{padding-bottom:var(--w-spacing, 8px);}
.pl-\\[var\\(--w-spacing\\,_8px\\)\\]{padding-left:var(--w-spacing, 8px);}
.pr-\\[var\\(--w-spacing\\,_8px\\)\\]{padding-right:var(--w-spacing, 8px);}
.pt-\\[var\\(--w-spacing\\,_8px\\)\\]{padding-top:var(--w-spacing, 8px);}"
`;

exports[`arbitrary padding with warp tokens works 1`] = `
"/* layer: default */
.p-\\[--w-spacing\\]{padding:var(--w-spacing);}
.px-\\[--w-spacing\\]{padding-left:var(--w-spacing);padding-right:var(--w-spacing);}
.py-\\[--w-spacing\\]{padding-top:var(--w-spacing);padding-bottom:var(--w-spacing);}
.pb-\\[--w-spacing\\]{padding-bottom:var(--w-spacing);}
.pl-\\[--w-spacing\\]{padding-left:var(--w-spacing);}
.pr-\\[--w-spacing\\]{padding-right:var(--w-spacing);}
.pt-\\[--w-spacing\\]{padding-top:var(--w-spacing);}"
`;

exports[`arbitrary padding works 1`] = `
"/* layer: default */
.p-\\[8\\]{padding:0.8rem;}
.p-\\[8px\\]{padding:8px;}
.p-\\[8rem\\]{padding:8rem;}
.px-\\[8\\]{padding-left:0.8rem;padding-right:0.8rem;}
.px-\\[8px\\]{padding-left:8px;padding-right:8px;}
.px-\\[8rem\\]{padding-left:8rem;padding-right:8rem;}
.py-\\[8\\]{padding-top:0.8rem;padding-bottom:0.8rem;}
.py-\\[8px\\]{padding-top:8px;padding-bottom:8px;}
.py-\\[8rem\\]{padding-top:8rem;padding-bottom:8rem;}
.pb-\\[8\\]{padding-bottom:0.8rem;}
.pb-\\[8px\\]{padding-bottom:8px;}
.pb-\\[8rem\\]{padding-bottom:8rem;}
.pl-\\[8\\]{padding-left:0.8rem;}
.pl-\\[8px\\]{padding-left:8px;}
.pl-\\[8rem\\]{padding-left:8rem;}
.pr-\\[8\\]{padding-right:0.8rem;}
.pr-\\[8px\\]{padding-right:8px;}
.pr-\\[8rem\\]{padding-right:8rem;}
.pt-\\[8\\]{padding-top:0.8rem;}
.pt-\\[8px\\]{padding-top:8px;}
.pt-\\[8rem\\]{padding-top:8rem;}"
`;

exports[`margin works 1`] = `
"/* layer: default */
.-m-8{margin:-0.8rem;}
.m-8{margin:0.8rem;}
.m-auto{margin:auto;}
.mx-2{margin-left:0.2rem;margin-right:0.2rem;}
.my-4{margin-top:0.4rem;margin-bottom:0.4rem;}
.mb-8{margin-bottom:0.8rem;}
.ml-32{margin-left:3.2rem;}
.ml-auto{margin-left:auto;}
.mr-16{margin-right:1.6rem;}
.mt-16{margin-top:1.6rem;}"
`;

exports[`padding works 1`] = `
"/* layer: default */
.p-8{padding:0.8rem;}
.p-auto{padding:auto;}
.px-2{padding-left:0.2rem;padding-right:0.2rem;}
.py-4{padding-top:0.4rem;padding-bottom:0.4rem;}
.pb-8{padding-bottom:0.8rem;}
.pl-32{padding-left:3.2rem;}
.pl-auto{padding-left:auto;}
.pr-16{padding-right:1.6rem;}
.pt-16{padding-top:1.6rem;}"
`;

exports[`should not render styles for margin with invalid arbitrary values 1`] = `""`;

exports[`should not render styles for padding with invalid arbitrary values 1`] = `""`;
